src/com/google/ProductionInstrumentationReporter.java

```diff
@@ -0,0 +1,464 @@
+        String encodedDetails = lineItem.substring(lineItem.indexOf(':') + 1);
+
+        StringCharIterator encodedDetailsAsCharIt = new StringCharIterator(encodedDetails);
+
+        int[] temp = new int[5];
+        temp[0] = Base64VLQ.decode(encodedDetailsAsCharIt); // Index in list this.fileNames
+        temp[1] = Base64VLQ.decode(encodedDetailsAsCharIt); // Index in list this.functionNames
+        temp[2] = Base64VLQ.decode(encodedDetailsAsCharIt); // Index in list this.types
+        temp[3] = Base64VLQ.decode(encodedDetailsAsCharIt); // Line number
+        temp[4] = Base64VLQ.decode(encodedDetailsAsCharIt); // Column number
+
+        parameterMapping.putIfAbsent(unqiueParam, temp);
+      }
+
+    }
+
+    public String getFileName(String uniqueParam) {
+      return fileNames.get(parameterMapping.get(uniqueParam)[0]);
+    }
+
+    public String getFunctionName(String uniqueParam) {
+      return functionNames.get(parameterMapping.get(uniqueParam)[1]);
+    }
+
+    public InstrumentationType getType(String uniqueParam) {
+      return types.get(parameterMapping.get(uniqueParam)[2]);
+    }
+
+    public int getLineNo(String uniqueParam) {
+      return parameterMapping.get(uniqueParam)[3];
+    }
+
+    public int getColNo(String uniqueParam) {
+      return parameterMapping.get(uniqueParam)[4];
+    }
+
+    /**
+     * This function will return a list of unique parameters which match the criteria specified by
+     * the predicate. As an example, it can return a list of all unique parameters which match a
+     * specific file name.
+     */
+    private List<String> getAllMatchingValues(Predicate<String> comparisonPredicate) {
+      List<String> result = new ArrayList<>();
+      for (String key : parameterMapping.keySet()) {
+        if (comparisonPredicate.test(key)) {
+          result.add(key);
+        }
+      }
+      return result;
+    }
+
+  }
+
+  /**
+   * The instrumentation will send the report in a JSON format where the JSON is a dictionary of the
+   * encoded params to an object of the data collected. Initially this will just be the frequency.
+   */
+  private static class ProfilingData {
+
+    public int frequency;
+
+    public ProfilingData(int frequency) {
+      this.frequency = frequency;
+    }
+
+  }
+
+  /**
+   * This class contains a detailed breakdown of each instrumentation point. For any encoded param,
+   * this class will contain its details. The `executed` property will be a percent of total times
+   * the instrumentation point was called across all provided instrumentation reports. i.e if
+   * this.executed = 100, then every report would have this point present. Similarly, ProfilingData
+   * takes the average of all ProfilingData's for this instrumentation point in all reports.
+   */
+  private static class ProfilingResult {
+
+    public String param;
+    public InstrumentationType type;
+    public int lineNo;
+    public int colNo;
```

Comment:
If you introduce `Location` class that I suggested above - you won't need to have lineNo and colNo here and instead can just have `location` field I think?